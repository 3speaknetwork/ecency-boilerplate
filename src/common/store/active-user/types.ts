import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
