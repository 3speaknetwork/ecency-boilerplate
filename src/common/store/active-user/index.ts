import { Dispatch } from "redux";

import { Account } from "../accounts/types";
import { ActiveUser, Actions, ReloadAction, UpdateAction, ActionTypes } from "./types";

import * as ls from "../../util/local-storage";

const load = (): ActiveUser | null => {
  const name = ls.get("active_user");
  if (name && ls.get(`user_${name}`)) {
    return {
      username: name,
      data: { name },
    };
  }

  return null;
};

export const initialState: ActiveUser | null = typeof window !== "undefined" ? load() : null;

export default (state: ActiveUser | null = initialState, action: Actions): ActiveUser | null => {
  switch (action.type) {
    case ActionTypes.RELOAD: {
      return load();
    }
    case ActionTypes.UPDATE: {
      const { data } = action;
      return Object.assign({}, state, { data });
    }
    default:
      return state;
  }
};

/* Actions */
export const setActiveUser = (name: string | null) => async (dispatch: Dispatch) => {
  if (name) {
    ls.set("active_user", name);
  } else {
    ls.remove("active_user");
  }

  dispatch(reloadAct());
};

export const updateActiveUser = (data: Account) => async (dispatch: Dispatch) => {
  dispatch(updateAct(data));
};

/* Action Creators */
export const reloadAct = (): ReloadAction => {
  return {
    type: ActionTypes.RELOAD,
  };
};

export const updateAct = (data: Account): UpdateAction => {
  return {
    type: ActionTypes.UPDATE,
    data,
  };
};
