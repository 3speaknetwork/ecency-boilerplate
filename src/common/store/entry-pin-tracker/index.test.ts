import reducer, {initialState, fetchAct, setAct} from "./index";
import {logoutAct, loginAct} from "../active-user";

let state = initialState;

it("1- default state", () => {
    expect(state).toMatchSnapshot();
});

it("2- fetchAct", () => {
    state = reducer(state, fetchAct());
    expect(state).toMatchSnapshot();
});

it("3- setAct", () => {
    state = reducer(state, setAct(true));
    expect(state).toMatchSnapshot();
});

it("4- setAct", () => {
    state = reducer(state, setAct(false));
    expect(state).toMatchSnapshot();
});
