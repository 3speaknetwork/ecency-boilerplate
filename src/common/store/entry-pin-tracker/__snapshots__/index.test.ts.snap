// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "canFetch": true,
  "pinned": false,
}
`;

exports[`2- fetchAct 1`] = `
Object {
  "canFetch": false,
  "pinned": false,
}
`;

exports[`3- setAct 1`] = `
Object {
  "canFetch": true,
  "pinned": true,
}
`;

exports[`4- setAct 1`] = `
Object {
  "canFetch": true,
  "pinned": false,
}
`;
